Mathlib/RingTheory/MvPowerSeries/PiTopology.lean

```diff
@@ -150,44 +151,34 @@ theorem variables_tendsto_zero [Semiring R] :
   · simpa only [ite_eq_right_iff] using
       Eventually.of_forall fun x h' ↦ (not_exists.mp h x h').elim
 
-theorem tendsto_pow_zero_of_constantCoeff_nilpotent [CommSemiring R]
+theorem isTopologicallyNilpotent_of_constantCoeff_isNilpotent [CommSemiring R]
     {f} (hf : IsNilpotent (constantCoeff σ R f)) :
-    Tendsto (fun n : ℕ => f ^ n) atTop (nhds 0) := by
+    IsTopologicallyNilpotent f := by
   classical
   obtain ⟨m, hm⟩ := hf
-  simp_rw [tendsto_iff_coeff_tendsto, coeff_zero]
+  simp_rw [IsTopologicallyNilpotent, tendsto_iff_coeff_tendsto, coeff_zero]
   exact fun d ↦ tendsto_atTop_of_eventually_const fun n hn ↦
     coeff_eq_zero_of_constantCoeff_nilpotent hm hn
 
-theorem tendsto_pow_zero_of_constantCoeff_zero [CommSemiring R]
+theorem isTopologicallyNilpotent_of_constantCoeff_zero [CommSemiring R]
     {f} (hf : constantCoeff σ R f = 0) :
     Tendsto (fun n : ℕ => f ^ n) atTop (nhds 0) := by
-  apply tendsto_pow_zero_of_constantCoeff_nilpotent
+  apply isTopologicallyNilpotent_of_constantCoeff_isNilpotent
   rw [hf]
   exact IsNilpotent.zero
 
-/-- The powers of a `MvPowerSeries` converge to 0 iff its constant coefficient is nilpotent.
-N. Bourbaki, *Algebra II*, [bourbaki1981] (chap. 4, §4, n°2, corollaire de la prop. 3) -/
-theorem tendsto_pow_of_constantCoeff_nilpotent_iff [CommRing R] [DiscreteTopology R] (f) :
-    Tendsto (fun n : ℕ => f ^ n) atTop (nhds 0) ↔
-      IsNilpotent (constantCoeff σ R f) := by
-  refine ⟨?_, tendsto_pow_zero_of_constantCoeff_nilpotent⟩
-  intro h
-  suffices Tendsto (fun n : ℕ => constantCoeff σ R (f ^ n)) atTop (nhds 0) by
-    simp only [tendsto_def] at this
-    specialize this {0} _
-    suffices ∀ x : R, {x} ∈ nhds x by exact this 0
-    rw [← discreteTopology_iff_singleton_mem_nhds]; infer_instance
-    simp only [map_pow, mem_atTop_sets, ge_iff_le, Set.mem_preimage,
-      Set.mem_singleton_iff] at this
-    obtain ⟨m, hm⟩ := this
-    use m
-    apply hm m (le_refl m)
-  simp only [← @comp_apply _ R ℕ, ← tendsto_map'_iff]
-  simp only [Tendsto, map_le_iff_le_comap] at h ⊢
-  refine le_trans h (comap_mono ?_)
-  rw [← map_le_iff_le_comap]
-  exact Continuous.continuousAt (continuous_constantCoeff R)
+/-- Assuming the base ring has a discrete topology, the powers of a `MvPowerSeries` converge to 0
```

Comment:
It's probably impossible to prove it this way, because of the file circularity, but it seems to me that this is a particular case of that for linear topologies.

Reply:
Indeed my main motivation was the import tree, but I could also create a file just for these results and prove them in the "sensible" order. I think the only meaningful change is that `isTopologicallyNilpotent_of_constantCoeff_isNilpotent` would only hold for rings and not semirings, is that an issue ?

Reply:
Then this is not worth the pain. Can you just add a note on the top of the file that explains that the proof is done twice to avoid creating a new file? (The other option would be to have both at the same place.)